prose/html/post.page.tmpl

```diff
@@ -42,6 +42,10 @@
 
 <link rel="stylesheet" href="/syntax.css" />
 {{if .HasCSS}}<link rel="stylesheet" href="{{.CssURL}}" />{{end}}
+
+<script type="text/javascript" async
```

Comment:
Is there a way to determine if the markdown contains mathjax?  Ideally we would only load this JS if there is mathjax in the post.

This also breaks one of our features: "No javascript" so I'm not sure if this is going to work for us in its current state. What does this javascript do?

Reply:
@neurosnap takes the content rendered in the span block from the goldmark plugin and renders an svg of the mathematical equation based on it. example here: https://www.mathjax.org/

Reply:
The generated `span` tags are rendered with the `math` class, so it should be possible to check the generated HTML for that and conditionally include the JS.
I didn't find any way to server render the SVGs though, so this probably isn't possible without sending JS to the client.

Reply:
> I didn't find any way to server render the SVGs though, so this probably isn't possible without sending JS to the client.

I think this could be an interesting opportunity for us.  We should investigate how hard it would be to generate the SVGs in our markdown processor so we can avoid injecting JS into posts.

Reply:
Turns out Mathjax [supports generating SVGs on the server](https://docs.mathjax.org/en/latest/server/start.html), but only in NodeJS. The simplest way would probably be to also run a Node server in parallel, which would really not be the cleanest solution.
An alternative would be to replace Goldmark with [pandoc](https://pandoc.org/), which has a more feature-complete MD renderer. This is accessed as a CLI though, so building a small wrapper would be necessary.

Reply:
Another idea is to re-implement [the node lib](https://github.com/mathjax/MathJax-node/blob/master/lib/main.js) to golang and then use that.

I know that sounds like a lot of work but it could be a fun project to work on.  Obviously another issue would be keeping our golang implementation up-to-date with the official one.